### PR TITLE
Fix java benchmark bug, fix python lib cache

### PR DESCRIPTION
**Pull request by BSBandme**

### CODE REVIEW

File: benchmarks/java/pom.xml [hunk 1]
```diff
@@ -14,7 +14,10 @@
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
-      <version>3.5.0</version>
+	  <version>1.0</version>
```
Comment (xfxyjwf) [inline benchmarks/java/pom.xml]: This seems wrong?

File: benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java [hunk 2]
```diff
@@ -22,6 +23,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
+@VmOptions("-XX:-TieredCompilation")
```
Comment (xfxyjwf) [inline benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java]: Why do we need this?

Comment (BSBandme) [reply benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java]: It seems a bug for caliper that it set CICompilerCount to 1, for TieredCompilation this value must be at least 2. So I just disable TieredCompilation.

File: benchmarks/java/pom.xml (hunk 1 above)
Comment (BSBandme) [reply benchmarks/java/pom.xml]: Then what version number should I set?

Comment (BSBandme) [reply benchmarks/java/pom.xml]: I think the benchmark should use protobuf-java in the same repo not just for a specific version?

File: benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java (hunk 2 above)
Comment (xfxyjwf) [reply benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java]: So it fails on kokoro but not locally? Can we set some environment variable to make it also work on kokoro? I'm worried that disabling TieredCompilation would change the performance characteristics and make the benchmark result less meaningful.

File: benchmarks/java/pom.xml (hunk 1 above)
Comment (xfxyjwf) [reply benchmarks/java/pom.xml]: In that case, can you use something like "<version>${protobuf.version}</version>" here and pass "-Dprotobuf.version=..." when invoking mvn in the Makefile?

File: benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java (hunk 2 above)
Comment (BSBandme) [reply benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java]: I think it fail on kokoro is the version of java is different from local one. In fact caliper's fix after 1.0-beta-2 for that is just [disabling TieredCompilation](https://github.com/google/caliper/blob/97a906bd6d91e9e0a30d413966b3a11469cb9ebf/caliper-runner/src/main/java/com/google/caliper/runner/target/Jvm.java#L39).  I think it's OK to do that

File: benchmarks/java/pom.xml (hunk 1 above)
Comment (BSBandme) [reply benchmarks/java/pom.xml]: Done

File: benchmarks/java/pom.xml [hunk 3]
```diff
@@ -14,7 +14,10 @@
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
-      <version>3.5.0</version>
+	  <version>${protobuf.version}</version>
```
Comment (xfxyjwf) [inline benchmarks/java/pom.xml]: Fix the indentation.

File: benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java (hunk 2 above)
Comment (xfxyjwf) [reply benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java]: Please add comments for why this is needed and link the caliper source as well.

File: benchmarks/java/pom.xml (hunk 3 above)
Comment (BSBandme) [reply benchmarks/java/pom.xml]: Done

File: benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java (hunk 2 above)
Comment (BSBandme) [reply benchmarks/java/src/main/java/com/google/protobuf/ProtoCaliperBenchmark.java]: Done

**Pull request merged**